### PR TITLE
Prism xamarin working branch

### PR DESCRIPTION
**Pull request by LeeParrishMSFT**
Adding Prism templates for Xamarin projects. There are two versions, one that uses shared project, another that is .NET Standard library for the shared code. I also updated the FakeGenShell and FakeMSBuildProject to not add the compile items explicitly into the project file for the .NET Standard project as the default for .NET Standard projects is to compile all items within the directory and not have them added individually into the project file (.csproj).

### CODE REVIEW

**Review by jermws — COMMENTED** [review]
Will there always be a MainPage? e.g. await NavigationService.NavigateAsync("NavigationPage/MainPage");

File: templates/Xamarin/Projects/Default.Prism.NETStd/wts.DefaultProject.Android/wts.DefaultProject.Android.csproj [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    <AndroidResource Include="Resources\layout\Toolbar.axml" />
+    <AndroidResource Include="Resources\values\styles.xml" />
+  </ItemGroup>
+  <ItemGroup>
+    <None Include="Properties\AndroidManifest.xml" />
+  </ItemGroup>
+  <ItemGroup>
+    <ProjectReference Include="..\wts.DefaultProject\wts.DefaultProject.csproj">
+      <Project>{c84d6ab8-12fc-46e8-838e-30056ca6c432}</Project>
+      <Name>App4</Name>
```
Comment (milazzom) [inline templates/Xamarin/Projects/Default.Prism.NETStd/wts.DefaultProject.Android/wts.DefaultProject.Android.csproj]: I think the Name element value should be set to wts.DefaultProject.

File: templates/Xamarin/Projects/Default.Prism.NETStd/wts.DefaultProject.iOS/wts.DefaultProject.iOS.csproj [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    <Reference Include="Xamarin.iOS" />
+  </ItemGroup>
+  <ItemGroup>
+    <PackageReference Include="Xamarin.Forms" Version="2.5.0.280555" />
+    <PackageReference Include="Prism.Unity.Forms" Version="7.0.0.396" />
+  </ItemGroup>
+  <ItemGroup>
+    <ProjectReference Include="..\wts.DefaultProject\wts.DefaultProject.csproj">
+      <Project>{c84d6ab8-12fc-46e8-838e-30056ca6c432}</Project>
+      <Name>App4</Name>
```
Comment (milazzom) [inline templates/Xamarin/Projects/Default.Prism.NETStd/wts.DefaultProject.iOS/wts.DefaultProject.iOS.csproj]: Change <Name>App4</Name> to <Name>wts.DefaultProject</Name>

File: templates/Xamarin/Projects/Default.Prism.NETStd/wts.DefaultProject.Android/wts.DefaultProject.Android.csproj (hunk 1 above)
Comment (LeeParrishMSFT) [reply templates/Xamarin/Projects/Default.Prism.NETStd/wts.DefaultProject.Android/wts.DefaultProject.Android.csproj]: Yes, it should, though strange thing is that it compiles fine, I guess it's based on Project GUID, and the name is only for us "humans". I've corrected this on all the .NET Std Projects, UWP, iOS, and Android.

File: templates/Xamarin/Projects/Default.Prism.NETStd/wts.DefaultProject.iOS/wts.DefaultProject.iOS.csproj (hunk 2 above)
Comment (LeeParrishMSFT) [reply templates/Xamarin/Projects/Default.Prism.NETStd/wts.DefaultProject.iOS/wts.DefaultProject.iOS.csproj]: See above, fix went in for all three projects in the .NET Standard base project template.

**Review by milazzom — APPROVED** [review]
Looks good!

**Pull request merged**